src/Framework.RabbitMq.Consumer/BackgroundServices/RabbitMqBackgroundService.cs

```diff
@@ -0,0 +1,129 @@
+using System.Text;
+
+using Framework.RabbitMq.Consumer.Interfaces;
+using Framework.RabbitMq.Consumer.Settings;
+
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using RabbitMQ.Client;
+
+namespace Framework.RabbitMq.Consumer.BackgroundServices;
+
+public class RabbitMqBackgroundService : BackgroundService
+{
+    private readonly IRabbitMqClient _client;
+
+    private readonly IRabbitMqConsumerInitializer _consumerInitializer;
+
+    private readonly IDeadLetterRabbitMqAuditService? _deadLetterAuditService;
+
+    private readonly ILogger<RabbitMqBackgroundService> _logger;
+
+    private readonly IRabbitMqMessageProcessor _messageProcessor;
+
+    private readonly IProcessedMessageRabbitMqAuditService? _processedAuditService;
+
+    private readonly RabbitMqSettings _settings;
+
+    private IModel? _channel;
+
+    private IConnection? _connection;
+
+    public RabbitMqBackgroundService(
+        IRabbitMqClient client,
+        IRabbitMqConsumerInitializer consumerInitializer,
+        IRabbitMqMessageProcessor messageProcessor,
+        IProcessedMessageRabbitMqAuditService? processedAuditService,
+        IDeadLetterRabbitMqAuditService? deadLetterAuditService,
+        IOptions<RabbitMqSettings> options,
+        ILogger<RabbitMqBackgroundService> logger)
+    {
+        this._client = client;
+        this._consumerInitializer = consumerInitializer;
+        this._messageProcessor = messageProcessor;
+        this._processedAuditService = processedAuditService;
+        this._deadLetterAuditService = deadLetterAuditService;
+        this._logger = logger;
+        this._settings = options.Value;
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        this._connection = await this._client.TryConnectAsync();
+        this._channel = this._connection!.CreateModel();
+        this._consumerInitializer.Initialize(this._channel);
+
+        await this.Listen(stoppingToken);
+    }
+
+    private async Task Listen(CancellationToken token)
+    {
+        this._logger.LogInformation(
+            "Listening RabbitMQ events has started on {Host}:{Port}",
+            this._settings.Server.Host,
+            this._settings.Server.Port);
+
+        while (!token.IsCancellationRequested)
+        {
+            await Delay(this._settings.Consumer.ReceiveMessageDelayMilliseconds, token);
```

Comment:
Delay must apply only for empty queue. Or you can get infinity process period for long queue